### DISCUSSION TITLE
Linux GPIO Sysfs vs libgpiod

### DISCUSSION THREAD

**Discussion by teddylambert** [nasa/fprime#1236] [Ideas]
I'm working on a project that will be interacting with the GPIO pins on a Linux board (running OpenWRT Linux, based on kernel 4.14). I was reading up on Linux GPIO interfacing and looking at the F-Prime Linux GPIO Driver component, and noticed that all of the component interfacing is based on sysfs. However, sysfs seems to be deprecated since kernel 4.8, and libgpiod has become the de-facto standard and more popular way. Are there any plans to update the driver to libgpiod? Or are there compatibility issues or reasonings for keeping the F-Prime driver using sysfs?

I'm leaning towards implementing libgpiod for the future reusability and support benefits, and the libgpiod C library looks to make the code pretty simple. Interested to hear other peoples' thoughts though.

**Comment by LeStarch** [comment]
We've had this suggested before. However, since we have a working solution, we haven't had the motivation to change.  I don't recall any specific notes against gpiod except that isn't what is currently implemented.
